When not logged in, on dandiset landing page: TypeError: Cannot read property 'admin' of null

```
instrument.js:109 Error evaluating async computed property: TypeError: Cannot read property 'admin' of null
    at https://gui.dandiarchive.org/js/app.ceb36f81.js:1:110535
    at l (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:38:38932)
    at Generator._invoke (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:38:40262)
    at Generator.next (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:38:39361)
    at r (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:4:50911)
    at s (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:4:51114)
    at https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:4:51173
    at new Promise (<anonymous>)
    at https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:4:51054
    at a.get (https://gui.dandiarchive.org/js/app.ceb36f81.js:1:110824)
    at nr.get (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:30684)
    at nr.evaluate (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:31685)
    at a._async_computed$userCanModifyDandiset (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:33378)
    at a.<anonymous> (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:3589)
    at nr.get (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:30684)
    at new nr (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:30602)
    at a.t.$watch (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:34115)
    at x (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:100:36530)
    at a.created (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:100:35920)
    at ne (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:11677)
    at Nn (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:29145)
    at a.t._init (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:34510)
    at new a (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:35770)
    at on (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:21670)
    at init (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:20536)
    at n (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:21872)
    at p (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:44201)
    at d (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:43830)
    at b (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:44815)
    at d (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:43945)
    at b (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:44815)
    at d (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:43945)
    at a.__patch__ (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:48311)
    at a.t._update (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:27008)
    at a.r (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:27814)
    at nr.get (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:30684)
    at new nr (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:30602)
    at In (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:27838)
    at a.kr.$mount (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:65645)
    at init (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:20547)
    at p (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:44201)
    at d (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:43830)
    at b (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:44815)
    at d (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:43945)
    at b (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:44815)
    at d (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:43945)
    at a.__patch__ (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:48311)
    at a.t._update (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:27008)
    at a.r (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:27814)
    at nr.get (https://gui.dandiarchive.org/js/chunk-vendors.4dc60169.js:10:30684)
```

--- a/web/src/views/DandisetLandingView/DandisetLandingView.vue
+++ b/web/src/views/DandisetLandingView/DandisetLandingView.vue
@@ -162,6 +162,10 @@ export default {
   asyncComputed: {
     userCanModifyDandiset: {
       async get() {
+        if (!this.user) {
+          return false;
+        }
+
         if (this.user.admin) {
           return true;
         }

